src/js/actions/tools.js

```diff
@@ -59,8 +59,13 @@ define(function (require, exports) {
             document.layers.all.first().isBackground) {
             return descriptor.PlayObject(defaultObj);
         }
-        var layerSpec = document.layers.allSelected.toList(),
-            layerRef = layerSpec
+        var layerSpec = document.layers.allSelected.toList();
+
+        if (layerSpec.isEmpty()) {
+            return descriptor.PlayObject(defaultObj);
```

Comment:
Also, why do we pass in layer references here if it suffices to just pass in the document reference?